llvm/docs/LangRef.rst

```diff
@@ -19363,27 +19363,27 @@ vector <N x eltty>, imm is a signed integer constant in the range
 -N <= imm < N. For a scalable vector <vscale x N x eltty>, imm is a signed
 integer constant in the range -X <= imm < X where X=vscale_range_min * N.
 
-'``llvm.experimental.stepvector``' Intrinsic
+'``llvm.stepvector``' Intrinsic
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-This is an overloaded intrinsic. You can use ``llvm.experimental.stepvector``
+This is an overloaded intrinsic. You can use ``llvm.stepvector``
 to generate a vector whose lane values comprise the linear sequence
 <0, 1, 2, ...>. It is primarily intended for scalable vectors.
 
 ::
 
-      declare <vscale x 4 x i32> @llvm.experimental.stepvector.nxv4i32()
-      declare <vscale x 8 x i16> @llvm.experimental.stepvector.nxv8i16()
+      declare <vscale x 4 x i32> @llvm.stepvector.nxv4i32()
+      declare <vscale x 8 x i16> @llvm.stepvector.nxv8i16()
 
-The '``llvm.experimental.stepvector``' intrinsics are used to create vectors
+The '``llvm.stepvector``' intrinsics are used to create vectors
 of integers whose elements contain a linear sequence of values starting from 0
 with a step of 1.  This experimental intrinsic can only be used for vectors
 with integer elements that are at least 8 bits in size. If the sequence value
```

Comment:
Why does this 8 bit limitation exist?

Reply:
I believe that this has been like that from the beginning when this intrinsic has been added https://github.com/llvm/llvm-project/commit/748ae5281d4f7f0ff261ba9e8c57e6b6fcfdb31e.

I guess the limitation comes from the fact that at that time there was no discussions about native support for scalable vectors of types smaller that i8, but @david-arm maybe remembers more.

Reply:
It was a bit of a struggle to find the original patch in Phabricator because you cannot search it. You can see the discussion on the patch here - https://reviews.llvm.org/D97299. There were concerns in both a community call and in comments on the patch that stepvector would be essentially meaningless for vectors of i1 types. Code generation for types smaller than i8 may require additional complexity, and until there is a genuine use case for it we decided to avoid adding support for it.

Reply:
However, there isn't really a fundamental reason why we can't support if someone has a use case that they need to support.